pkg/errors/gatherer.go

```diff
@@ -44,6 +47,32 @@ func (g *Gatherer) Failed() <-chan struct{} {
 	return g.failed
 }
 
+// DoneOrFailed closes the returned channel when either:
+// all routines returned successfully OR
+// any routine returned with an error.
+func (g *Gatherer) DoneOrFailed() <-chan struct{} {
```

Comment:
Make it return nothing, call the new `WaitDoneOrFailed()` as suggested. Rename to `WaitDoneOrFailed()`.

Reply:
done